service/src/main/java/bio/terra/cbas/initialization/InstanceInitializer.java

```diff
@@ -0,0 +1,24 @@
+package bio.terra.cbas.initialization;
+
+import bio.terra.cbas.util.BackfillOriginalWorkspaceIdService;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.stereotype.Component;
+
+@Component
+public class InstanceInitializer implements ApplicationListener<ContextRefreshedEvent> {
+
+  @Autowired BackfillOriginalWorkspaceIdService backfillOriginalWorkspaceIdService;
```

Comment:
I think it should be generally preferred to make this `private final` provided via the constructor (which makes it auto-auto-wired). Then you shouldn't need the explicit autowired annotation
```suggestion
  private final BackfillOriginalWorkspaceIdService backfillOriginalWorkspaceIdService;
```